src/cmd/9pfuse/fuse.c

```diff
@@ -837,6 +840,32 @@ mountfuse(char *mtpt)
 		}
 	}
 
+	/* MacFUSE >=4 dropped support for passing fd */
+	if (strcmp(v, "macfuse") == 0) {
```

Comment:
@dancrossnyc MacFUSE >=3.3 supports the new mechanism but with different environment variables. My assumption is that since the VFS name `macfuse` is new, `v` will only be set to `macfuse` on >=4. I branch on that; all other VFS names (and therefore older versions) should continue to the old code below the conditional.